### ISSUE TITLE
Connected wallet disappears on reload page

### ISSUE THREAD

**Issue by ilya2207** [WalletConnect/web3modal#822]
**Describe the bug**
When connect to wallet and reload page first 2 logs give a connected account but further i get address `undefined` and isConnected `false`. This behavior reproduced in 2.0.0-rc.2, in 2.0.0-rc.1 it's work correctly. If press connect after reload page, account will connect automatically

**SDK Version**
 -  @web3modal/react - 2.0.0-rc.2
 - @web3modal/ethereum - 2.0.0-rc.2
 - ethers - ^5.7.2
 - wagmi - ^0.10.10

**To Reproduce**
1. Connect wallet
2. Reload page
3. Press again connect to wallet or log connected account
4. If you press connected wallet you will see qrcode or undefined in console,

**Expected behavior**
Wallet should connected

**Screenshots**
![image](https://user-images.githubusercontent.com/62308020/212269652-b0be50a4-d721-4974-aad1-6f4dc10b4294.png)

**Desktop (please complete the following information):**
 - OS: Windows
 - Browser Chrome
 - Version 108.0.5359.125

**Comment by 0xAsimetriq** [comment]
Do you have `autoConnect` set to true in wagmi client configuration?
https://github.com/WalletConnect/web3modal/blob/V2/examples/react/src/pages/_app.tsx#L20

If yes, will need to see a reproduction code / repo of the setup to dive deeper.

**Comment by ilya2207** [comment]
Yes i have

**Comment by ilya2207** [comment]
https://codesandbox.io/s/stupefied-alex-9j9psp?file=/src/WalletConnectWrapper.jsx

**Comment by 0xAsimetriq** [comment]
Ty for quick repro, I think this issue could be related to codesanbox editor. I forked it and visited page as a normal url (i.e. full page) and everything seems to work

1. I connected
2. Refreshed the page few times and address was there

Similarly, this same flow works on many other dapps, so issue has to be specific to setup / environment.


https://user-images.githubusercontent.com/3154053/212275373-9574dc50-cb9e-4afe-b907-f87d58af5e37.mov
